### ISSUE TITLE
The tyre model's MC Dropout is unseeded at inference, so measured numbers are not reproducible

### ISSUE THREAD

**Issue by VforVitorio** [VforVitorio/F1_Strat_Manager#735]
Found by the A-2 gate while auditing #726. It is not a defect of that PR — it is older and wider, and it affects **published numbers**.

## The defect

N26 samples MC Dropout at inference and never seeds it:

- `src/agents/tire_agent.py:1094` — `model.train()  # keep dropout active for MC`, **no seed**
- `src/strategy/eval/tire_holdout.py:225-226` — `model.train()  # keep dropout active for MC sampling` followed by **`torch.manual_seed(seed)`**

Same operation, near-identical comment, one seeded and one not. **The eval twin got the seed; the production inference path never did.** This repo's dominant defect class, in its purest form.

## Measured

Five identical runs of Lusail lap 33 through `run_lap(profile="no-llm")`:

```
run 0: p10=2.60 p50=2.80 p90=3.00 deg=0.7240
run 1: p10=2.60 p50=2.80 p90=3.00 deg=0.7240
run 2: p10=2.60 p50=2.80 p90=3.10 deg=0.7240
run 3: p10=2.60 p50=2.80 p90=3.10 deg=0.7240
run 4: p10=2.60 p50=2.80 p90=3.00 deg=0.7240
```

`laps_to_cliff_p90` moves between runs. Most of the variance is hidden by rounding the quantiles to one decimal — which is why nobody noticed — but not all of it. The gate measured a STAY_OUT score moving **0.493** on one lap between two identical captures.

## Why it matters more than it looks

The Monte Carlo seeds its own RNG at 42 and shares draws across candidates, so the layer is deterministic **downstream** of a stochastic **upstream**. The consequences:

1. **A published claim in #726 is not reproducible.** The "Lusail argmax STAY_OUT 54/57 → 52/57" census came out 53/57 and 52/57 on two identical captures at the same commit. One realization of a stochastic measurement was written into a commit message as a property of the code. Corrected on that issue.
2. **The `f1-eval decision-modes` headline numbers carry unmeasured run-to-run variance** — the 65% decline rate, the −2.23 lap signed bias, the 40/198 coverage. Nobody knows their error bar.
3. **Two identical race states can produce different recommendations**, which contradicts the determinism the rest of the layer is built for (seeded RNG, common random numbers across candidates, byte-identical goldens).

## The fix, and the question attached to it

The mechanical part is small: seed at the same seam the eval twin already does. **Prefer importing or sharing that code rather than adding a second `torch.manual_seed` call** — a third copy of the same decision is how this family of bug reproduces.

The judgement part is not mechanical and should be decided deliberately: **should inference be seeded at all?** MC Dropout exists to express model uncertainty, and a fixed seed makes that uncertainty identical on every call for a given input. That is what you want for a reproducible decision layer and arguably not what you want from an uncertainty estimate. The eval twin already chose "seeded"; production should make the same choice explicitly rather than by omission.

## Acceptance

- [ ] Decide seeded-vs-unseeded for inference, and write the reason down where the next reader will find it
- [ ] If seeded: one source shared with `tire_holdout`, not a second call site
- [ ] A test asserting two identical calls return identical `TireOutput` — none exists today, which is the mechanism
- [ ] Re-run `f1-eval decision-modes` and state whether the published 65% / −2.23 figures move
